src/Directory.Build.props

```diff
@@ -5,6 +5,7 @@
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>..\NUnitAdapter.snk</AssemblyOriginatorKeyFile>
     <CodeAnalysisRuleSet>..\..\Osiris.Extended.ruleset</CodeAnalysisRuleSet>
+    <TreatWarningsAsErrors Condition="'$(Configuration)' == 'Release'">true</TreatWarningsAsErrors>
```

Comment:
The reason this is conditional is that it's too annoying to be in the middle of writing something in the IDE and try to run or debug a test and get a build failure over something that is only a warning. Often there is an unused variable or incomplete formatting. These things shouldn't make it into the master branch, but they also definitely shouldn't slow down inner-loop development.